vmdb/app/controllers/application_controller.rb

```diff
@@ -1026,18 +1015,16 @@ def get_reports_menu(group=session[:group], tree_type="reports", mode="menu")
     else
       # Building custom reports array for super_admin/admin roles, it doesnt show up on menu if their menu was set which didnt contain custom folder in it
       temp = Array.new
-      subfolder = Array.new
-      rep = Array.new
-      @custom_folder = Array.new
+      subfolder = %w{ Custom }
+      @custom_folder = [ @sb[:grp_title] ]
       @custom_folder.push(@sb[:grp_title])
-      subfolder.push("Custom")
       @custom_folder.push([subfolder]) unless @custom_folder.include?([subfolder])
-      custom = MiqReport.all.sort{|a,b| a.rpt_type + a.filename.to_s + a.name <=> b.rpt_type  + b.filename.to_s + b.name}
-      custom.each do |r|
-        if r.rpt_type == "Custom" && (user.admin_user? || r.miq_group_id.to_i == session[:group].to_i)
-          rep.push(r.name) unless rep.include?(r.name)
-        end
-      end
+
+      custom = MiqReport.all.sort_by { |a| a.rpt_type + a.filename.to_s + a.name }
```

Comment:
I believe it will. I did some testing on @Fryguy's PR where he did it this way. @Fryguy : what do you think?